sonar-text-plugin/src/main/java/org/sonar/plugins/secrets/api/SecretMatcher.java

```diff
@@ -33,34 +34,61 @@ public class SecretMatcher {
   private final AuxiliaryPatternMatcher auxiliaryPatternMatcher;
   private final Predicate<InputFileContext> preFilter;
   private final Predicate<String> postFilter;
+  private final DurationStatistics durationStatistics;
 
-  SecretMatcher(Rule rule, PatternMatcher patternMatcher, AuxiliaryPatternMatcher auxiliaryPatternMatcher,
-    Predicate<InputFileContext> preFilter, Predicate<String> postFilter) {
+  SecretMatcher(Rule rule,
+    PatternMatcher patternMatcher,
+    AuxiliaryPatternMatcher auxiliaryPatternMatcher,
+    Predicate<InputFileContext> preFilter,
+    Predicate<String> postFilter,
+    DurationStatistics durationStatistics) {
     this.rule = rule;
     this.patternMatcher = patternMatcher;
     this.auxiliaryPatternMatcher = auxiliaryPatternMatcher;
     this.preFilter = preFilter;
     this.postFilter = postFilter;
+    this.durationStatistics = durationStatistics;
   }
 
-  public static SecretMatcher build(Rule rule) {
+  /**
+   * Creates a new SecretMatcher from provided Rule
+   * @param rule rule to extract matcher logic from
+   * @param durationStatistics instance to collect performance statistics
+   * @return a new SecretMatcher
+   */
+  public static SecretMatcher build(Rule rule, DurationStatistics durationStatistics) {
     PatternMatcher patternMatcher = PatternMatcher.build(rule.getDetection().getMatching());
     Predicate<InputFileContext> preFilter = PreFilterFactory.createPredicate(rule.getDetection().getPre());
     Predicate<String> postFilter = PostFilterFactory.createPredicate(rule.getDetection().getPost(), rule.getDetection().getMatching());
     AuxiliaryPatternMatcher auxiliaryMatcher = AuxiliaryPatternMatcherFactory.build(rule.getDetection().getMatching());
-    return new SecretMatcher(rule, patternMatcher, auxiliaryMatcher, preFilter, postFilter);
+    return new SecretMatcher(rule, patternMatcher, auxiliaryMatcher, preFilter, postFilter, durationStatistics);
   }
 
   public List<Match> findIn(InputFileContext fileContext) {
-    if (!preFilter.test(fileContext)) {
+    boolean isRejectedOnPreFilter = durationStatistics.timed(
```

Comment:
Consider providing methods like `timedPreFilter`, timedSuffixMatcher`, timedSuffixPost` etc. 
IMO will improve readability and will not leak `DurationStatistics.SUFFIX_PRE` and other constants to client code.